pbh-verifier/src/PBHVerifier.sol

```diff
@@ -0,0 +1,165 @@
+import {IWorldIDGroups} from "@world-id-contracts/interfaces/IWorldIDGroups.sol";
+import "@BokkyPooBahsDateTimeLibrary/BokkyPooBahsDateTimeLibrary.sol";
+
+contract PBHVerifier {
+    using ByteHasher for bytes;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  ERRORS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Thrown when attempting to reuse a nullifier
+    error InvalidNullifier();
+    
+    /// @notice Thrown when the provided external nullifier year doesn't
+    /// match the current year
+    error InvalidExternalNullifierYear();
+    
+    /// @notice Thrown when the provided external nullifier month doesn't
+    /// match the current month
+    error InvalidExternalNullifierMonth();
+    
+    /// @notice Thrown when the provided external 
+    /// nullifier pbhNonce >= numPbhPerMonth
+    error InvalidPbhNonce();
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Events                                ///
+    //////////////////////////////////////////////////////////////////////////////
+    
+    /// @notice Emitted when a verifier is updated in the lookup table.
+    ///
+    /// @param nullifierHash The nullifier hash that was used.
+    event PBH(
+        uint256 indexed nullifierHash
+    );
+    
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Structs                               ///
+    //////////////////////////////////////////////////////////////////////////////
+    
+    /**
+    * User Operation struct
+    * @param sender                - The sender account of this request.
+    * @param nonce                 - Unique value the sender uses to verify it is not a replay.
+    * @param initCode              - If set, the account contract will be created by this constructor/
+    * @param callData              - The method call to execute on this account.
+    * @param accountGasLimits      - Packed gas limits for validateUserOp and gas limit passed to the callData method call.
+    * @param preVerificationGas    - Gas not calculated by the handleOps method, but added to the gas paid.
+    *                                Covers batch overhead.
+    * @param gasFees               - packed gas fields maxPriorityFeePerGas and maxFeePerGas - Same as EIP-1559 gas parameters.
+    * @param paymasterAndData      - If set, this field holds the paymaster address, verification gas limit, postOp gas limit and paymaster-specific extra data
+    *                                The paymaster will pay for the transaction instead of the sender.
+    * @param signature             - Sender-verified signature over the entire request, the EntryPoint address and the chain ID.
+    */
+    struct PackedUserOperation {
```

Comment:
Can we inherit this type from the `account-abstraction` contracts?

Reply:
Haha yeah for sure :) sorry this isn't even close to being ready for review haha. I just opened the PR to show the external nullifier implementation!